src/shims/mod.rs

```diff
@@ -27,9 +28,74 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         // There are some more lang items we want to hook that CTFE does not hook (yet).
         if this.tcx.lang_items().align_offset_fn() == Some(instance.def.def_id()) {
-            // FIXME: return a real value in case the target allocation has an
-            // alignment bigger than the one requested.
-            let n = u128::max_value();
+            let ptr = this.read_scalar(args[0])?.not_undef()?;
+            let align = this.read_scalar(args[1])?.not_undef()?;
+
+            let n = {
+                let p = this.force_bits(ptr, this.pointer_size())?;
```

Comment:
this will error before doing the stride >= a check, so even if it could pessimize to `u128::max_value()` it would yield a real value. Move it into the `if` body (or remove it if the below trick works)

Reply:
Your trick seems to work so I removed this :P